01_homework/hw4_task9b.cpp

```diff
@@ -0,0 +1,112 @@
+#include <opencv2/core.hpp>
+#include <opencv2/imgproc.hpp>
+#include <opencv2/highgui.hpp>
+#include <iostream>
+#include <future>
+
+using namespace std;
+using namespace cv;
+
+constexpr int angle_step = 5;
+
+void show_wait_destroy(const char* winname, cv::Mat img) {
+    imshow(winname, img);
+    moveWindow(winname, 500, 0);
+    waitKey(0);
+    destroyWindow(winname);
+}
+
+Mat get_diagonal_line_mask(int line_size, int angle) {
+    double angle_rad = ( angle * M_PI ) / 180.0;
+
+    // Create mask of proper size
+    Mat test_image(abs((line_size * cos(angle_rad))) + 1,
+                   abs((line_size * sin(angle_rad))) + 1,
+                   CV_8U);
+
+    // Create rotated rectagle and width = 0
+    RotatedRect rRect = RotatedRect(Point2f(abs((line_size/2 * sin(angle_rad))),
+                                            abs((line_size/2 * cos(angle_rad)))),
+                                    Size2f(0, line_size),
+                                    angle);
```

Comment:
praca domowa z matematyki odrobiona 👍 
Wydaje mi się, że openCV ma funkcję rotate() i może po prostu zrobić Mat z linią poziomą na środku i za pomocą rotate ją obrócić i chyba mniej kodu by było. Taka sugestia tylko.

Reply:
Znalazlem w dokumentacji OpenCV tylko takie rotate() co obraca od wielokrotnosci 90stopni. Mało przydatne w tym przypadku :(

Reply:
Aha, to chyba takie kawałek kodu by trzeba było napisać: http://opencvexamples.blogspot.com/2014/01/rotate-image.html
Ale sam już uznaj, czy warto, czy nie :)